Generation/src/components/ParticleGunAlg.cpp

```diff
@@ -0,0 +1,85 @@
+
+#include "ParticleGunAlg.h"
+
+#include "boost/tokenizer.hpp"
+
+#include "GaudiKernel/DeclareFactoryEntries.h"
+#include "GaudiKernel/RndmGenerators.h"
+
+#include "HepMC/GenEvent.h"
+
+//-----------------------------------------------------------------------------
+// Implementation file for class : ParticleGun
+//
+// 2008-05-18 : Patrick Robbe
+// 2014-06-23 : Benedikt Hegner (simplification for Gaudi)
+//-----------------------------------------------------------------------------
+
+/// Declaration of the Algorithm Factory
+DECLARE_ALGORITHM_FACTORY( ParticleGunAlg )
+
+/// Standard constructor, initializes variables
+ParticleGunAlg::ParticleGunAlg( const std::string& name,
+                          ISvcLocator* pSvcLocator)
+  : GaudiAlgorithm ( name , pSvcLocator ) {
+  // Generation Method
+  declareProperty ( "ParticleGunTool" ,
+                    m_particleGunTool ) ;
+  declareProperty ( "VertexSmearingToolPGun" ,
+                     m_vertexSmearingTool ) ;
+  declarePrivateTool(m_vertexSmearingTool, "FlatSmearVertex/VertexSmearingToolPGun");
+  declarePrivateTool(m_particleGunTool, "MomentumRangeParticleGun/ParticleGunTool");
+  declareOutput("hepmc", m_hepmchandle);
+}
+
+/// Destructor
+ParticleGunAlg::~ParticleGunAlg() {}
+
+/// Initialisation. Check parameters
+StatusCode ParticleGunAlg::initialize() {
+  StatusCode sc = GaudiAlgorithm::initialize( ) ; // Initialize base class
+  if ( sc.isFailure() ) return sc ;
+  debug() << "==> Initialise" << endmsg ;
+  return StatusCode::SUCCESS;
+}
+
+/// Main execution
+StatusCode ParticleGunAlg::execute() {
+  StatusCode sc = StatusCode::SUCCESS ;
+  Gaudi::LorentzVector theFourMomentum ;
+  Gaudi::LorentzVector origin ;
+  int thePdgId ;
```

Comment:
Please add a comment that this is actually retrieved from the particleGunTool. Also maybe put it directly next to the function that sets it.